packages/agent/app/create-harness.ts

```diff
@@ -47,30 +51,48 @@ function *runTest(parentFrame: ParentFrame, test: TestImplementation, context: T
     try {
       console.debug('[harness] running step', step);
       parentFrame.send({ type: 'step:running', path: stepPath });
-      let result: TestContext | void = yield step.action(context);
+
+      let result: TestContext | void = yield timebox(step.action(context), 2000)
+
       if (result != null) {
         context = {...context, ...result};
       }
       parentFrame.send({ type: 'step:result', status: 'ok', path: stepPath });
     } catch(error) {
       console.error('[harness] step failed', step, error);
-      parentFrame.send({ type: 'step:result', status: 'failed', error: serializeError(error), path: stepPath });
+      if (error.name === 'TimeoutError') {
+        parentFrame.send({
+          type: 'step:result',
+          status: 'failed',
+          timeout: true,
+          path: stepPath
+        })
+      } else {
```

Comment:
Can we do this without using exceptions for control flow? We could implement this as two operations racing against each other. Though we'd need an Effection equivalent to `Promise.race`.